doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +122,80 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
 Configuring your application
 ****************************
 
-If your application uses a provided library or targets a specific board, you might want to change the default configuration of the library or board.
+You might want to change the default options of the application.
 There are different ways of doing this, but not all will store your configuration permanently.
 
-The default configuration for a library is specified in its :file:`Kconfig` file.
-Similarly, the default configuration for a board is specified in its :file:`*_defconfig` file (and its :file:`Kconfig.defconfig` file, see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
-The configuration for your application, which might override some default options of the libraries or the board, is specified in a :file:`prj.conf` file in the application directory.
+Zephyr's configuration system is distributed, and so is the |NCS| one.
+This means that there are multiple sources of configuration.
+In the |NCS|, the configuration system consists of the following elements:
+
+* Kconfig configuration files that focus on software-related options.
+* Devicetree specification (DTS) configuration files that focus on hardware-related options.
+* Partition manager configuration files that focus on memory layout.
+
+Software-related configuration
+==============================
+
+The software-related configuration comes from Kconfig files (called simply :file:`Kconfig`).
+There are several layers that provide the configuration options:
+
+* The lowest layer consists of defaults provided in Kconfig files of all system components, libraries, and application itself.
+* The defaults are overlayed by the options specified by the selected board in its :file:`*_defconfig` file and its :file:`Kconfig.defconfig` file (see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
+* The configuration is additionally overlayed by the :file:`prj.conf` file provided by the application.
+
+After merging the configuration sets imported from these configuration files, the build system creates a single :file:`.config` file that describes the entire software configuration of the system.
+This is done once and configuration comes solely from the :file:`.config` file until the project is reevaluated.
+
+.. note::
+   You can edit the :file:`.config` file and rebuild the application to check how the configuration changes.
+   The changes will be lost when the application project is reevaluated.
```

Comment:
What does `reevaluated` means ?

I'm not familiar with that term when building Zephyr / NCS.

Reply:
I meant when you rerun cmake, which will cause build files to be reconstructed. Don't know how else to name it.

Reply:
Can you use exactly that? "Rerun CMake" or "rebuild the files"? "reevaluated" is not used in our docs and that's one of the points I wanted to discuss with you on Teams.

Reply:
> I meant when you rerun cmake, which will cause build files to be reconstructed.

wrong, the `.config` is kept when you re-run CMake, but will be lost if you remove the build folder / do a pristine build.
https://docs.zephyrproject.org/latest/application/index.html#rebuilding-an-application

> Alternatively, enter one of the following commands to delete all generated files, including the .config files that contain the application’s current configuration information for those board types.
> 
> `west build -t pristine`
> or
> `ninja pristine`

Reply:
"wrong, the .config is kept when you re-run CMake, but will be lost if you remove the build folder / do a pristine build."
... or modify one of the Kconfig fragments used, e.g. prj.conf.

Reply:
Rephrased, pending final verification.

Reply:
>  ... or modify one of the Kconfig fragments used, e.g. prj.conf.

Right, thanks for adding that.